Lesson4/main_L4.py

```diff
@@ -0,0 +1,217 @@
+    new_nickname = StringField('New nickname', validators=[DataRequired(), Length(3, 30)])
+    new_password = StringField('New password', validators=[DataRequired(), Length(3, 30)])
+
+    new_category = RadioField('Choose your Hogwarts House:', validators=[InputRequired(message=None)],
+                          choices=[('Griffindor', 'Griffindor'),
+                                   ('Ravenclaw', 'Ravenclaw'),
+                                   ('Hufflepuff', 'Hufflepuff'),
+                                   ('Slytherin', 'Slytherin')])
+
+    submit = SubmitField('Update User Info')
+
+
+usr_present = False
+
+
+@app.route("/", methods=["GET", "POST"])
+@app.route("/login", methods=["GET", "POST"])
+def login():
+
+    global usr_present
+    l_form = LoginForm()
+    # Create table
+    sqlite_manager_L4.create_table()
+
+    if l_form.validate_on_submit():
+        user.nickname = l_form.nickname.data
+        user.password = l_form.password.data
+
+        # Check DB
+        usr_present = sqlite_manager_L4.check_user(user.nickname, user.password)
+
+        # SUBMIT
+        if l_form.submit.data:
+            if usr_present:
+                # Redirect to /magic. Show user from DB
+                return redirect(url_for('magic_item'))
+            else:
+                # No such User in DB to Submit. Error Raise
+                raise Exception("User is absent. Try to Register.")  # +
+        # REGISTER
+        elif l_form.register.data:
+            if usr_present:
+                # Treat Existing user as login.To /magic
+                return redirect(url_for('magic_item'))
+            else:
+                # To New User registration
+                return redirect(url_for('register'))
+
+    return render_template('login_form.html', form=l_form)
+
+
+@app.route("/registration", methods=["GET", "POST"])
+def register():
+    global usr_present
+    r_form = RegisterForm()
+
+    if r_form.validate_on_submit():
+        user.nickname = r_form.nickname.data
+        user.password = r_form.password.data
+
+        # User that differs from /login but present in DB check
+        usr_present = sqlite_manager_L4.check_user(user.nickname, user.password)
+
+        if usr_present:
+            # Trying to register Existing User
+            raise Exception("Nickname is taken. Try another one.")
+        else:
+            # To house selection
+            return redirect(url_for('hogwarts_house'))
+
+    return render_template('register_form.html', form=r_form)
+
+
+@app.route("/house", methods=["GET", "POST"])
+def hogwarts_house():
+    h_form = HogwartsHouseForm()
+
+    if h_form.validate_on_submit():
+        user.house = h_form.category.data
+        # To Last Page:
+        return redirect(url_for('magic_item'))
+
+    return render_template('hogwarts_house_L4.html', form=h_form, nickname=user.nickname, password=user.password)
+
+
+@app.route("/item", methods=["GET", "POST"])
+def magic_item():
+    global usr_present
```

Comment:
key word `global` is not needed if you not updated the global context but using it